src/main/java/net/sf/jabref/model/database/BibDatabase.java

```diff
@@ -173,12 +184,36 @@ public synchronized BibEntry getEntryByKey(String key) {
      * Inserts the entry, given that its ID is not already in use.
      * use Util.createId(...) to make up a unique ID for an entry.
      *
-     * @param entry the entry to insert into the database
+     * @param entry BibEntry to insert
      * @return false if the insert was done without a duplicate warning
      * @throws KeyCollisionException thrown if the entry id ({@link BibEntry#getId()}) is already  present in the database
      */
     public synchronized boolean insertEntry(BibEntry entry) throws KeyCollisionException {
-        return this.insertEntry(entry, false);
+        return insertEntry(entry, EntryEventLocation.ALL);
+    }
+
+    /**
+     * Inserts the entry, given that its ID is not already in use.
+     * use Util.createId(...) to make up a unique ID for an entry.
+     *
+     * @param entry BibEntry to insert
+     * @param eventLocation Event location which is affected by inserting a new entry
+     * @return false if the insert was done without a duplicate warning
+     */
+    public synchronized boolean insertEntry(BibEntry entry, EntryEventLocation eventLocation)
```

Comment:
It looks like this is an almost copy of the other insertEntry method. https://github.com/obraliar/jabref/blob/c948c623fa212b29169f31f64b848fc1bf3bb9e8/src/main/java/net/sf/jabref/model/database/BibDatabase.java#L228
Probably it is best to add `undo` as an event location.